client.go

```diff
@@ -920,20 +926,30 @@ func (c *Client[TTx]) distributeJob(job *rivertype.JobRow, stats *JobStatistics)
 	}
 
 	var event *Event
-	switch job.State {
-	case rivertype.JobStateCancelled:
-		event = &Event{Kind: EventKindJobCancelled, Job: job, JobStats: stats}
-	case rivertype.JobStateCompleted:
-		event = &Event{Kind: EventKindJobCompleted, Job: job, JobStats: stats}
-	case rivertype.JobStateScheduled:
-		event = &Event{Kind: EventKindJobSnoozed, Job: job, JobStats: stats}
-	case rivertype.JobStateAvailable, rivertype.JobStateDiscarded, rivertype.JobStateRetryable, rivertype.JobStateRunning:
-		event = &Event{Kind: EventKindJobFailed, Job: job, JobStats: stats}
-	case rivertype.JobStatePending:
-		panic("completion subscriber unexpectedly received job in pending state, river bug")
-	default:
-		// linter exhaustive rule prevents this from being reached
-		panic("unreachable state to distribute, river bug")
+
+	if job != nil {
+		switch job.State {
+		case rivertype.JobStateCancelled:
+			event = &Event{Kind: EventKindJobCancelled, Job: job, JobStats: stats}
+		case rivertype.JobStateCompleted:
+			event = &Event{Kind: EventKindJobCompleted, Job: job, JobStats: stats}
+		case rivertype.JobStateScheduled:
+			event = &Event{Kind: EventKindJobSnoozed, Job: job, JobStats: stats}
+		case rivertype.JobStateAvailable, rivertype.JobStateDiscarded, rivertype.JobStateRetryable, rivertype.JobStateRunning:
+			event = &Event{Kind: EventKindJobFailed, Job: job, JobStats: stats}
+		case rivertype.JobStatePending:
+			panic("completion subscriber unexpectedly received job in pending state, river bug")
+		default:
+			// linter exhaustive rule prevents this from being reached
+			panic("unreachable state to distribute, river bug")
+		}
+	} else {
+		switch queue.PausedAt {
+		case nil:
+			event = &Event{Kind: EventKindQueueResumed, Queue: queue}
+		default:
+			event = &Event{Kind: EventKindQueuePaused, Queue: queue}
```

Comment:
What do you think about just having the producer generate the event rather than having a separate field that's then reassembled into an event post-hoc? See:

https://github.com/riverqueue/river/pull/328/files

Feels a little bit cleaner, and it may help for assembling a full queue object.